packages/internal/bridge/bridge-sample-app/src/deposit.ts

```diff
@@ -27,7 +38,141 @@ async function deposit() {
+
+    approvalRes!.unsignedTx.gasLimit = BridgeMethodsGasLimit.DEPOSIT_SOURCE;
+    approvalRes!.unsignedTx.nonce = approvalNonce;
+    approvalRes!.unsignedTx.gasPrice = approvalGasPrice.mul(2);
+
+    console.log('approvalRes.unsignedTx');
+    console.log(approvalRes!.unsignedTx);
+
+    console.log('signing approval');
+    const approvalTxSig = await params.rootWallet.signTransaction(approvalRes!.unsignedTx);
+    console.log('approvalTxSig', approvalTxSig);
+
+    const sendApprovalRes = await params.rootWallet.provider.sendTransaction(approvalTxSig);
+    console.log('sendApprovalRes', sendApprovalRes);
+
+    await waitForReceipt(sendApprovalRes.hash, params.rootProvider);
+  } else {
+    console.log('no approval required');
+  }
+
+  const depositReq: BridgeTxRequest = {
+    senderAddress: params.sender,
+    recipientAddress: params.recipient,
+    token: params.sepoliaToken,
+    amount: params.amount,
+    sourceChainId: ETH_SEPOLIA_CHAIN_ID,
+    destinationChainId: ZKEVM_TESTNET_CHAIN_ID,
+    gasMultiplier: 1.1,
+  }
+
+  console.log('depositReq', depositReq)
+  let depositRes: BridgeTxResponse;
+  try {
+    depositRes = await tokenBridge.getUnsignedBridgeTx(depositReq);
+    console.log('depositRes', depositRes);
+  } catch(err) {
+    console.log('depositErr', err);
+    return
+  }
+
+  if (!depositRes!.unsignedTx) {
+    console.log('unable to generate deposit tx');
+    return
+  }
+
+  const depositNonce = await params.rootWallet.getTransactionCount();
+  const depositGasPrice = await params.rootProvider.getGasPrice();
+
+  depositRes!.unsignedTx.gasLimit = BridgeMethodsGasLimit.DEPOSIT_SOURCE;
+  depositRes!.unsignedTx.nonce = depositNonce;
+  depositRes!.unsignedTx.gasPrice = depositGasPrice.mul(2);
+
+  depositRes!.unsignedTx.value = ethers.BigNumber.from(depositRes!.unsignedTx.value);
+
+  console.log('depositRes.unsignedTx');
+  console.log(depositRes!.unsignedTx);
+
+  console.log('signing deposit');
+  const depositTxSig = await params.rootWallet.signTransaction(depositRes!.unsignedTx);
+  console.log('depositTxSig', depositTxSig);
+
+  const sendDepositRes = await params.rootWallet.provider.sendTransaction(depositTxSig);
+  console.log('sendDepositRes', sendDepositRes);
+
+  await waitForReceipt(sendDepositRes.hash, params.rootProvider);
+  
+  console.log('Deposit submitted txHash:',sendDepositRes.hash);
+
+  const txStatusReq:TxStatusRequest = {
+    sourceChainId: ETH_SEPOLIA_CHAIN_ID,
+    transactions: [{
+      txHash: sendDepositRes.hash
+    }]
+  }
+
+  let complete:boolean = false;
+  let attempts = 0;
+  while(!complete) {
+    attempts++;
+    const txStatusRes: TxStatusResponse = await tokenBridge.getTransactionStatus(txStatusReq);
+    console.log(`TxStatusResponse attempt: ${attempts}`);
+    console.log(util.inspect(txStatusRes, {showHidden: false, depth: null, colors: true}));
+    if (txStatusRes.transactions[0].status === StatusResponse.COMPLETE) {
```

Comment:
should we check to see if the transaction is in a failed state where retrying won't help? 
should we have a limit on the number of attempts?